src/posix.mak

```diff
@@ -444,7 +444,7 @@ $G/backend.a: $(G_OBJS) $(SRC_MAKE)
 	$(AR) rcs $@ $(G_OBJS)
 
 $G/lexer.a: $(LEXER_SRCS) $(LEXER_ROOT) $(HOST_DMD_PATH) $(SRC_MAKE)
-	CC="$(HOST_CXX)" $(HOST_DMD_RUN) -lib -of$@ $(MODEL_FLAG) -J$G -L-lstdc++ $(DFLAGS) $(LEXER_SRCS) $(LEXER_ROOT)
+	CC="$(HOST_CXX)" $(HOST_DMD_RUN) -lib -of$@ $(MODEL_FLAG) -J$G -J../res -L-lstdc++ $(DFLAGS) $(LEXER_SRCS) $(LEXER_ROOT)
```

Comment:
The additional -J switch appears to have nothing to do with this PR.

Reply:
The circular import adds extra module dependencies on the bare bones "lexer" library. I have been making other PRs to reduce the effect of this, but ultimately will probably move the vendor field back to globals for now.

Reply:
Ok. The rest of the PR looks good to me.

Reply:
Can we remove the lexer library from the makefiles? We already have the Dub package.